src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.Unix.cs

```diff
@@ -68,26 +87,38 @@ private SocketConfig GetSocketConfig(IPAddress address, byte[] buffer, int timeo
         private Socket GetRawSocket(SocketConfig socketConfig)
         {
             IPEndPoint ep = (IPEndPoint)socketConfig.EndPoint;
-
-            // Setting Socket.DontFragment and .Ttl is not supported on Unix, so socketConfig.Options is ignored.
             AddressFamily addrFamily = ep.Address.AddressFamily;
+
             Socket socket = new Socket(addrFamily, SocketType.Raw, socketConfig.ProtocolType);
             socket.ReceiveTimeout = socketConfig.Timeout;
             socket.SendTimeout = socketConfig.Timeout;
+            if (addrFamily == AddressFamily.InterNetworkV6)
+            {
+                socket.DualMode = false;
```

Comment:
Why is this necessary?  Isn't it only set to true by the two-arg Socket ctor that's not being used here?

Reply:
Ah, nevermind, I guess it's necessary because it's actually calling setsockopt.